agent/api/ecsclient/client.go

```diff
@@ -279,6 +279,52 @@ func (client *APIECSClient) getCustomAttributes() []*ecs.Attribute {
 	return attributes
 }
 
+func (client *APIECSClient) buildContainerStateChangePayload(change api.ContainerStateChange) *ecs.ContainerStateChange {
+	statechange := &ecs.ContainerStateChange{
+		ContainerName: &change.ContainerName,
+	}
+
+	if change.Reason != "" {
+		if len(change.Reason) > ecsMaxReasonLength {
+			trimmed := change.Reason[0:ecsMaxReasonLength]
+			statechange.Reason = &trimmed
+		} else {
+			statechange.Reason = &change.Reason
+		}
+	}
+	stat := change.Status.String()
+	if stat == "DEAD" {
+		stat = "STOPPED"
+	}
+
+	if stat != "STOPPED" && stat != "RUNNING" {
```

Comment:
you can compare directly with `api.ContainerStopped` and `api.ContainerRunning` without converting the status to string.